src/lib.rs

```diff
@@ -8,123 +8,166 @@ pub struct DB<T> {
-pub struct DBViewMut<T> {
-    entries: Vec<&mut T>,
+pub struct DBViewMut<'a, T: 'a> {
+    entries: Vec<&'a mut T>,
 }
 
 /// Filters a DBView using the the given predicate.
-pub fn filter_one<T, F>(view: &DBView<T>, predicate: F) -> DBView< T>
-    where F: Fn(&T) -> bool
+pub fn filter_one<'a, 'b: 'a, T, F>(view: &'a DBView<'b, T>, predicate: F) -> DBView<'b, T>
+    where F: for<'c> Fn(&'c T) -> bool
 {
-    unimplemented!()
+    view.select_where(predicate)
 }
 
 /// Filters two DBView structs using the same predicate, producing two separate results. This is
 /// the moral equivalent of doing the two filters separately.
-pub fn filter_two<T, F>(view_a: &DBView<T>,
-                        view_b: &DBView<T>,
+pub fn filter_two<'a, 'b, T, F>(view_a: &DBView<'a, T>,
+                        view_b: &DBView<'b, T>,
                         predicate: F)
-                        -> (DBView<T>, DBView<T>)
-    where F: Fn(&T) -> bool
+                        -> (DBView<'a, T>, DBView<'b, T>)
+    where F: for<'c> Fn(&'c T) -> bool
 {
-    unimplemented!()
+    // couldn't get this to work like filter_one, or by calling filter_one
+    // Couldn't figure out the lifetime hierarchy
+    (DBView::<'a, T>{entries: view_a.entries
+                          .clone()
+                          .into_iter()
+                          .filter(|&x| predicate(x))
+                          .collect()},
+     DBView::<'b, T>{entries: view_b.entries
+                          .clone()
+                          .into_iter()
+                          .filter(|&x| predicate(x))
+                          .collect()})
 }
 
 impl<T> DB<T> {
     /// Creates a DB from the given list of entries
     pub fn new(data: Vec<T>) -> DB<T> {
-        unimplemented!()
+        DB::<T>{data: data}
     }
 
     /// Creates a new DBView containing all entries in `self` which satisfy `predicate`
-    pub fn select_where<F>(&self, predicate: F) -> DBView<T>
-        where F: Fn(&T) -> bool
+    pub fn select_where<'a, F>(&'a self, predicate: F) -> DBView<'a, T>
+        where F: for<'c> Fn(&'c T) -> bool
     {
-        unimplemented!()
+        self.as_view().select_where(predicate)
     }
 
     /// Creates a new DBView containing all entries in `self` which satisfy `predicate`
-    pub fn select_where_mut<F>(&mut self, predicate: F) -> DBViewMut<T>
-        where F: Fn(&T) -> bool
+    pub fn select_where_mut<'a, F>(&'a mut self, predicate: F) -> DBViewMut<'a, T>
+        where F: for<'c> Fn(&'c T) -> bool
     {
-        unimplemented!()
+        self.as_view_mut().select_where_mut(predicate)
     }
 
     /// Returns a DBView consisting on the entirety of `self`
-    pub fn as_view(&self) -> DBView<T> {
-        unimplemented!()
+    pub fn as_view<'a>(&'a self) -> DBView<'a, T> {
+        let mut new_view = DBView::<'a, T>{entries: Vec::new()};
+        for record in & self.data {
+            new_view.entries.push(record);
+        }
+
+        new_view
```

Comment:
This implementation could be cleaned up, similar to your implementations below which use the Iterator interface.